iOS/Controllers/MapViewController.cs

```diff
@@ -567,23 +572,29 @@ private void MapView_GeoViewDoubleTapped(object sender, GeoViewInputEventArgs e)
                         var roomMarker = new PictureMarkerSymbol(new RuntimeImage(mapPin));
                         roomMarker.OffsetY = uiImagePin.Size.Height * 0.65;
 
-                        // Create graphic
-                        var mapPinGraphic = new Graphic(GeometryEngine.LabelPoint(idResults.GeoElements.First().Geometry as Polygon), roomMarker);
+                        // Create graphic
+                        var mapPinGraphic = new Graphic(idResults.GeoElements.FirstOrDefault().Geometry.Extent.GetCenter(), roomMarker);
 
                         // Add pin to mapview
                         var graphicsOverlay = this.MapView.GraphicsOverlays["PinsGraphicsOverlay"];
                         graphicsOverlay.Graphics.Clear();
-                        graphicsOverlay.Graphics.Add(mapPinGraphic);
+                        graphicsOverlay.Graphics.Add(mapPinGraphic);
 
                         // Get room attribute from the settings. First attribute should be set as the searcheable one
-                        var roomAttribute = AppSettings.CurrentSettings.ContactCardDisplayFields[0];
-                        var employeeNameAttribute = AppSettings.CurrentSettings.ContactCardDisplayFields[1];
-                        var roomNumber = idResults.GeoElements.First().Attributes[roomAttribute];
-                        var employeeName = idResults.GeoElements.First().Attributes[employeeNameAttribute];
+                        var roomAttribute = AppSettings.CurrentSettings.ContactCardDisplayFields[0];
+                        var roomNumber = idResults.GeoElements.First().Attributes[roomAttribute];
 
                         if (roomNumber != null)
-                        {
-                            var employeeNameLabel = employeeName ?? string.Empty;
+                        {
+                            var employeeNameLabel = string.Empty;
+                            if (AppSettings.CurrentSettings.ContactCardDisplayFields.Count > 1)
+                            {
+
+                                var employeeNameAttribute = AppSettings.CurrentSettings.ContactCardDisplayFields[1];
+                                var employeeName = idResults.GeoElements.First().Attributes[employeeNameAttribute];

```

Comment:
The cached idResults.GeoElements.FirstOrDefault() can be used here if it were stored in a variable.